generators/component/hoc.js.hbs

```diff
@@ -0,0 +1,12 @@
+// @flow
+/**
+ *
+ * {{ properCase name }}
+ *
+ */
+
+const {{ properCase name}} = Component => {
```

Comment:
This template is not enough. Have minimum imports. I.e React, types definitions etc. 
Rename Component as WrappedComponent.

Template should looks like below: 
```
import React from 'react';

import getDisplayName from '../../core/utils/getDisplayName';

const {{ properCase name}} = WrappedComponent => {
  // ...and returns another component...
  class extends React.Component {
    constructor(props) {
      super(props);
      this.state = {};
    }

    render() {
    
      return <WrappedComponent {...this.props} />;
    }
  };
  {{ properCase name}}.displayName = `{{ properCase name}}(${getDisplayName(WrappedComponent)})`;
  return {{ properCase name}};
}
``` 

Also create a file at core/utils with following content. 
function getDisplayName(WrappedComponent) {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
}